google/oauth2/service_account.py

```diff
@@ -806,13 +920,17 @@ def _refresh_with_iam_endpoint(self, request):
             additional_claims={"scope": "https://www.googleapis.com/auth/iam"},
         )
         jwt_credentials.refresh(request)
+
+        headers = self._get_trust_boundary_header()
```

Comment:
you will get NotImplementedError here, thrown from ln 863?

Reply:
Also, there is no access token in this cred to call trust boundary endpoint

Reply:
I reverted the changes to idtoken as it turns out it doesn't need trust boundaries. Thanks for flagging it!